impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterFactory.java

```diff
@@ -93,6 +98,11 @@ public static KafkaCluster create(KafkaClusterConfig clusterConfig) {
         }
     }
 
+    private static KafkaClusterExecutionMode getExecutionMode(KafkaClusterConfig clusterConfig) {
+        return KafkaClusterExecutionMode.convertClusterExecutionMode(System.getenv().get(TEST_CLUSTER_EXECUTION_MODE),
```

Comment:
not part of your changes but it looks a bit dangerous that it defaults it if the string is unknown, maybe better to explode with error if the user supplies value "CONTAINDERER"